### ISSUE TITLE
Type inheritance

### ISSUE THREAD

**Issue by Gummibeer** [spatie/schema-org#53]
https://github.com/spatie/schema-org#type-inheritance

Since the methods are callable and it's possible to know which methods are also available - wouldn't it be cool to have the methods from another type mentioned in the doc-tag?

Let's take the `LocalBusiness` in this package it extends the `Organization` but it should also extend `Place` this isn't possible without a massive refactoring but the place methods could be listed as `@method`.

```php
/**
 * @method publicAccess(bool $value)
 * ...
 */
class LocalBusiness extends Organization
{
    // ...
}
```

Is this something you want to implement or do you want this to keep it like it is?

**Comment by sebastiandedeyne** [comment]
How can we determine that a type needs to inherit from multiple other types? Last I checked, this wasn't possible with the RDFA file. Also in that readme section:

> The spec rdfa document that's used to generate this code uses single inheritance for the types

If this has changed, or I was wrong, I'm open for improvements here. In that case, maybe we could look into traits for sharing properties between traits.

**Comment by Gummibeer** [comment]
https://github.com/schemaorg/schemaorg/blob/e68d896a784d2b1d21425bee9834116db586124b/data/schema.rdfa#L74-L79
The `LocalBusiness ` has two childs of `rdfs:subClassOf`.

**Comment by sebastiandedeyne** [comment]
Instead of `@method`, I think `@mixin \Spatie\Schema\Place` is a cleaner solution. 

Is the actual inheritance still useful then? I'm don't like the idea of doing `extends` for one arbitrary type, and docblocks for the rest of them. I'd rather follow one strategy for everything.

**Comment by Gummibeer** [comment]
@sebastiandedeyne I've searched for `@mixin` but haven't found an official doc-page!? I've also tried it in PhpStorm and it doesn't auto-complete `Place` methods for me. 😕 

And yes, you are right - it would be better to do all the same way - in this case: php-doc

My idea would be:
* define methods like now in class eg `Place`
* use `@method` and `@see` in extending classes

This will keep the method description in the main class and make it accessible by `@see` in all extending ones and allows auto-complete by `@method`.

**Comment by Gummibeer** [comment]
@sebastiandedeyne Ok, my problem was that I've tested with `publicAccess` but this wasn't available on `Place` - it's fixed in #54 . Now `@mixin` seems 100% fine and also a lot easier instead of listing all methods.

**Comment by sebastiandedeyne** [comment]
Released in 2.0.0, thanks again!
